sweep/sweeper.go

```diff
@@ -323,11 +321,26 @@ func (s *UtxoSweeper) Start() error {
 	// not change from here on.
 	s.relayFeeRate = s.cfg.FeeEstimator.RelayFeePerKW()
 
-	// Register for block epochs to retry sweeping every block.
-	bestHash, bestHeight, err := s.cfg.ChainIO.GetBestBlock()
+	// We need to get the current best block so we can register for block
+	// epochs and retry sweeping every block. We should get a notification
+	// immediately if we don't provide any epoch.
+	var (
+		bestHeight int32
+		bestHash   *chainhash.Hash
+	)
+	bestBlockEpochs, err := s.cfg.Notifier.RegisterBlockEpochNtfn(nil)
```

Comment:
since we re-register for block epochs below, is this even needed? Could make the `collector` below always wait for first block to arrive, as it seems like its the only one using it

Reply:
I moved the wait for the first block into the `collector` as you suggested. I wasn't sure how the notifier would behave after the first nil request. But I looked into it and it seems like we get exactly what we want with only registering once.
So I removed the extra registration.